ast/policy.go

```diff
@@ -409,6 +428,25 @@ func (c *Comment) Equal(other *Comment) bool {
 	return c.Location.Equal(other.Location) && bytes.Equal(c.Text, other.Text)
 }
 
+func (c *Comment) exposeJSONFields(fields map[string]bool) {
+	// Note: this is not used for location since Comments have legacy JSON marshaling behavior
+	c.jsonFields = fields
+}
+
+func (c *Comment) MarshalJSON() ([]byte, error) {
+	// TODO: Comment has inconsistent JSON field names starting with an upper case letter.
+	data := map[string]interface{}{
+		"Text": c.Text,
+	}
+
+	// TODO: Comment Location is also always included for legacy reasons. jsonFields data is ignored.
+	if c.Location != nil {
```

Comment:
Nit: Is this check needed if the location is also always included ?

Reply:
Hmm, that's a good point. In order to leave this functionality as it is I have changed things around in ee243588ae9d6fd14960d7d4406fcea541bbd3dd. I think it's clear enough to note this inconsistency on the struct and leave out the no-op custom marshal function for now. I have left the tests in however.